src/Controller/Controller_Pdf_Queue.php

```diff
@@ -278,6 +279,27 @@ protected function get_queue_tasks( $entry, $form, $notifications = [] ) {
 
 		$queue_data = apply_filters( 'gfpdf_queue_initialise', [], $entry, $form );
 
+		/**
+		 * Check if Asynchronous notification is enabled for this form.
+		 * https://docs.gravityforms.com/gform_is_asynchronous_notifications_enabled/
+		 */
+		$is_asynchronous = gf_apply_filters(
+			[
+				'gform_is_asynchronous_notifications_enabled',
+				$form['id'],
+			],
+			false,
+			$notifications['event'],
+			$notifications['id'],
+			$form,
+			$entry,
+			[]
+		);
+
+		if ( $is_asynchronous ) {
+			return $queue_data;
```

Comment:
@jestonihpi looks like a good spot for it, considering we need to pass in the notification ID and event type being processed.

Couple things of note:

* `$queue_data` may not be empty if the `gfpdf_queue_initialise` filter is used and returns a value
* `$notifications['event']` is produce a PHP notice because the array key doesn't exist. `$notifications` is an array which, when looped over, will contain an individual notification config that includes the `event` key. Looking at the GF codebase, you should be able to take the event from the first notification configuration in this array. 
* `$notifications_to_send` should be an array of all Notification IDs that are sent, not a single ID. Like the above, this will produce a PHP notice. Use array_column to get the ID from all notifications.